source/index-management/tests/README.rst

```diff
@@ -45,4 +44,157 @@ Tests
+
+Case 3: Driver can successfully drop search indexes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+#. Create a collection with a randomly generated name (referred to as ``coll0``).
+#. Create a new search index on ``coll0`` with the following definition:
+
+  .. code:: typescript
+
+    {
+      name: 'test-search-index',
+      definition: {
+        mappings: { dynamic: false }
+      }
+    }
+
+#. Assert that the command does not error and the server responds with a success.
+#. Run ``coll0.listSearchIndexes()`` repeatedly every 5 seconds until the following condition is satisfied:
+   1. An index with the ``name`` of ``test-search-index`` is present.  This index is referred to as ``index``.
+   2. The ``latestDefinition`` property of the search index has the field ``queryable`` with a value of ``true``.
+
+#. Run a ``dropSearchIndexes`` on ``coll0``, using ``test-search-index`` for the name.
+#. Run ``coll0.listSearchIndexes()`` repeatedly every 5 seconds until ``listSearchIndexes`` returns an empty array.
+
+This test fails if it times out waiting for the deletion to succeed.
+
+Case 4: Driver can update a search index
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+#. Create a collection with a randomly generated name (referred to as ``coll0``).
+#. Create a new search index on ``coll0`` with the following definition:
+
+  .. code:: typescript
+
+    {
+      name: 'test-search-index',
+      definition: {
+        mappings: { dynamic: false }
+      }
+    }
+
+#. Assert that the command does not error and the server responds with a success.
+#. Run ``coll0.listSearchIndexes()`` repeatedly every 5 seconds until the following condition is satisfied:
+   1. An index with the ``name`` of ``test-search-index`` is present.  This index is referred to as ``index``.
+   2. The ``latestDefinition`` property of the search index has the field ``queryable`` with a value of ``true``.
+
+#. Run a ``updateSearchIndex`` on ``coll0``, using the following definition.
+  
+  .. code:: typescript
+
+    {
+      name: 'test-search-index',
+      definition: {
+        mappings: { dynamic: true }
+      }
+    }
+
+#. Assert that the command does not error and the server responds with a success.
+#. Run ``coll0.listSearchIndexes()`` repeatedly every 5 seconds until the following condition is satisfied:
+   1. An index with the ``name`` of ``test-search-index`` is present.  This index is referred to as ``index``.
+   2. The ``latestDefinition`` property of the search index has the field ``queryable`` with a value of ``true``.
+   3. The ``latestDefinition`` property of the search index has the field ``status`` with the value of ``READY``.
+  
+#. Assert that an index is present with the name ``test-search-index`` and the definition has a
+  property ``mappings``, whose value is ``{ dynamic: true }``
+
+Case 5: ``dropSearchIndex`` suppresses namespace not found errors
```

Comment:
I didn't see this in the spec. I'm ignoring [error with code `26`](https://github.com/mongodb/mongo/blob/master/src/mongo/base/error_codes.yml)

Reply:
Nothing to change. It's just me who didn't read the spec as carefully as I should have.

It's [in the spec](https://github.com/mongodb/specifications/blob/78554de6a78fc996fa8d1ef9733b123a29ad8309/source/index-management/index-management.rst#namespacenotfound-errors).

> Drivers MUST suppress NamespaceNotFound errors for the dropSearchIndex helper.